compiler/src/dotty/tools/repl/ScriptEngine.scala

```diff
@@ -0,0 +1,77 @@
+package dotty.tools
+package repl
+
+import java.io.{Reader, StringWriter};
+import javax.script.{AbstractScriptEngine, Bindings, ScriptContext, ScriptEngine => JScriptEngine, ScriptEngineFactory, ScriptException, SimpleBindings}
+
+class ScriptEngine(factory: ScriptEngine.Factory) extends AbstractScriptEngine {
```

Comment:
Some documentation on what this is, how it works and how to use it would be helpful.